crates/hir_def/src/nameres/collector.rs

```diff
@@ -357,7 +358,7 @@ impl DefCollector<'_> {
                 }
 
                 i += 1;
-                if i == FIXED_POINT_LIMIT {
+                if i == FIXED_POINT_LIMIT.inner() {
```

Comment:
```suggestion
                if FIXED_POINT_LIMIT.check(i).is_err() {
```

Reply:
this changes semantics a bit though

Reply:
What do you mean? An off-by-one? It's incremented above though.

Reply:
It basically raises the limit by `1`, since we want to stop when `i` reaches the limit, but `check` can only error if `i` is `>` than the limit

Reply:
I mean.. other places already do `LIMIT.check(i + 1)`, but it doesn't seem too important. I'd go for bailing out when we go above the limit.